src/i2c/dma.rs

```diff
@@ -0,0 +1,624 @@
+use core::{marker::PhantomData, mem::transmute};
+
+use super::{I2c, Instance};
+use crate::dma::{
+    config::DmaConfig,
+    traits::{Channel, DMASet, PeriAddress, Stream},
+    ChannelX, MemoryToPeripheral, PeripheralToMemory, Transfer,
+};
+
+use nb;
+
+#[non_exhaustive]
+pub enum Error {
+    I2CError(super::Error),
+    TransferError,
+}
+
+/// Callback type to notify user code of completion I2C transfers
+pub type I2cCompleteCallback = fn(Result<(), Error>);
+
+pub trait I2CMasterWriteDMA {
+    /// This function is unsafe because user must ensure that `bytes` will live until `callback` called
+    unsafe fn write_dma(
+        &mut self,
+        addr: u8,
+        bytes: &[u8],
+        callback: Option<I2cCompleteCallback>,
```

Comment:
Is `Option` needed here.
As I understand `|_| {}` is equivalent to `None` here.